packages/compose-tunnel-agent/src/metadata.ts

```diff
@@ -0,0 +1,34 @@
+import fs from 'fs'
+import { Logger } from '@preevy/common'
+import { merge } from 'lodash'
+import { inspect } from 'util'
+
+export const envMetadata = async ({ env, log }: { env: NodeJS.ProcessEnv; log: Logger }) => {
+  const jsons = [
+    ...await Promise.all(
+      (env.ENV_METADATA_FILES || '')
```

Comment:
ENV_METADATA_FILES might be a bit of a problem if the folder contains 'space' because of the split. 
I think it might worth using a ENV_METADATA_DIR: 
can contain git metadata as well, but it can scan all the json files in the folder.

Alternatively, we can use a single merged json with sections. (can be passed either as a single file, or json var)
(in this case, it would be ENV_METADATA_FILE or ENV_METADATA)

Reply:
Also, we can support both metadata injections mechanisms as it can make it more user-friendly to consume and test, but for preevy cli it might make it simpler to use the same mechanism for git and cloud metadata.
I also we have the joined type of "EnvMetadata", but I'm not sure where we use this.

Reply:
Re pre-merged JSON suggestion: The driver metadata is generated at machine creation and saved on the machine persistent storage, so merging it at the `up` command would require reading it from the remote machine.

Re the directory suggestion: merge order is significant. I considered using a directory, but the order is not guaranteed. We can sort the directory listing and use filenames like `00-git.json`, `10-driver.json` but that requires knowing the order in advance, when generating the contents, or re-ordering the files JIT, again requiring reading the file from the remote machine.

Re spaces in filenames: I'll add an option to use a JSON array in the file list var, should solve this.

Reply:
> we have the joined type of "EnvMetadata", but I'm not sure where we use this.

It's the output of the `envMetadata` function, mostly to ensure the expected schema.